pubmed_parser/pubmed_oa_parser.py

```diff
@@ -87,6 +87,33 @@ def parse_article_meta(tree):
     return dict_article_meta
 
 
+def parse_date(tree, date_type):
+    """Parse publication dates based on the provided date type."""
+    def get_text(node):
+        return node.text if node is not None else None
+
+    pub_date_path = f".//pub-date[@pub-type=\"{date_type}\"]"
+    return {part: get_text(tree.find(f"{pub_date_path}/{part}")) for part in ["year", "month", "day"]}
+
+
+def format_date(date_dict):
+    """Format date dictionary to a string in the format day-month-year."""
+    day = date_dict.get("day") or "01"
+    month = date_dict.get("month") or "01"
+    year = date_dict.get("year") or ""
+
+    if year:
+        return f"{day}-{month}-{year}"
+    else:
+        return f"{day}-{month}"
+
+
+def get_year_as_int(date_dict):
+    """Get the year as an integer from the date dictionary."""
+    year = date_dict.get("year")
+    return int(year) if year is not None else None
```

Comment:
Na, we don't need a function for this. This is too small a unit.
Let's just use
```
try:
    ppub_year = int(date_dict["year"])
except KeyError:
    ppub_year = None
```